maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/AbstractSurefireMojo.java

```diff
@@ -2995,6 +2996,36 @@ private void resolveJUnitVintageEngine( Set<Artifact> providerArtifacts )
             Set<Artifact> resolvedArtifacts = resolveArtifact( null, junitVintageEngine ).getArtifacts();
             providerArtifacts.addAll( resolvedArtifacts );
         }
+
+        private void alignJUnitPlatformLauncher( Set<Artifact> providerArtifacts )
+        {
+            Map<String, Artifact> providerArtifactMap = new HashMap<String, Artifact>();
+            for ( Artifact artifact : providerArtifacts )
+            {
+                String key = artifact.getGroupId() + ":" + artifact.getArtifactId();
+                providerArtifactMap.put( key, artifact );
+            }
+            Artifact defaultLauncher = providerArtifactMap.get( "org.junit.platform:junit-platform-launcher" );
+            Artifact junitPlatformCommons = getProjectArtifactMap().get( "org.junit.platform:junit-platform-commons" );
+
+            if ( junitPlatformCommons.getVersion().equals( defaultLauncher.getVersion() ) )
+            {
+                return;
```

Comment:
Why not negation and then pass the code into IF body. Why return. The `new DefaultArtifact(` is afterwards.

Reply:
- Handle trivial and special cases up-front.
- Afert that the actual work code is free of nested/complicated `if/else` clauses.
- Keep indentation level low.